pinot-common/src/main/java/org/apache/pinot/common/lineage/SegmentLineageUtils.java

```diff
@@ -53,4 +54,19 @@ public static void filterSegmentsBasedOnLineageInPlace(Set<String> segments, Seg
       }
     }
   }
+
+  public static Set<String> getDuplicateSegmentsToFilterOut(SegmentLineage segmentLineage) {
```

Comment:
minor nit: Do we want to rename this to replacedSegments? duplicateSegments might not justify the API purpose. wdyt?

Reply:
Yes, good point. Replaced segments don't necessarily mean duplicates, made this over-simplification while implementing by mistake :)